### DISCUSSION TITLE
Can't handle ctrl-c signal

### DISCUSSION THREAD

**Discussion by shellphy** [laravel/framework#46305] [Q&A]
- Laravel Version: 10.1.15
- PHP Version: 8.1.2
- Database Driver & Version: none

### Description:

I use [php-webdriver](https://github.com/php-webdriver/php-webdriver) 's RemoteWebDriver to start my chrome but when I press ctrl-c to stop my script, my chrome is still open. so I try to call $driver->quit() function by register pcntl_signal function in my test.php script, it works fine.

```php
<?php

require __DIR__ . '/vendor/autoload.php';

use Facebook\WebDriver\Chrome\ChromeOptions;
use Facebook\WebDriver\Remote\RemoteWebDriver;
use Facebook\WebDriver\Remote\DesiredCapabilities;

class Test
{
    private $driver;

    public function registerSignal()
    {
        declare(ticks = 1);
        pcntl_signal(SIGINT, function ($signo) {
            if ($signo == SIGINT) {
                echo PHP_EOL.'ctrl+c'.PHP_EOL;
                exit;
            }
        });
    }

    public function __construct() 
    {
        $this->registerSignal();
    }

    public function __destruct()
    {
        $this->driver->quit();    
    }

    public function run()
    {
        $chromeOptions = new ChromeOptions();
        $capabilities = DesiredCapabilities::chrome();
        $capabilities->setCapability(ChromeOptions::CAPABILITY, $chromeOptions);
        $this->driver = RemoteWebDriver::create('http://localhost:4444', $capabilities);
        $this->driver->get('https://www.google.com');
        while (true) {
            sleep(1);
            // todo something
        }
    }
}

(new Test())->run();
```

But when I integrate this method to laravel command, it appears "Segmentation fault" at the point of ` $this->driver->quit();`.

**Comment by JurianArie** [comment]
You could refactor it to a command, they have signal handling.

https://laravel.com/docs/10.x/artisan#signal-handling

**Reply by williamdes** [reply]
Based on my tests for console commands it works best if you implement the `SignalableCommandInterface`.
See: https://stackoverflow.com/a/70122775/5155484
